esphome/components/adc/adc_sensor.cpp

```diff
@@ -16,7 +16,9 @@ void ADCSensor::set_attenuation(adc_attenuation_t attenuation) { this->attenuati
 
 void ADCSensor::setup() {
   ESP_LOGCONFIG(TAG, "Setting up ADC '%s'...", this->get_name().c_str());
+#if !(defined(USE_ADC_SENSOR_VCC) && defined(ARDUINO_ARCH_ESP8266))
```

Comment:
why this long check and not just 
```suggestion
#ifndef USE_ADC_SENSOR_VCC
```

Reply:
#if !(defined(USE_ADC_SENSOR_VCC) && defined(ARDUINO_ARCH_ESP8266))
looked cleaner for me because of the check for ESP8266 in https://github.com/esphome/esphome/blob/31f9e47d865ae11917a09dcbf85863e13eb0f5b7/esphome/components/adc/adc_sensor.cpp#L81
however here is only the #ifdef USE_ADC_SENSOR_VCC check: https://github.com/esphome/esphome/blob/31f9e47d865ae11917a09dcbf85863e13eb0f5b7/esphome/components/adc/adc_sensor.cpp#L4 and USE_ADC_SENSOR_VCC is only possible on ESP8266: https://github.com/esphome/esphome/blob/31f9e47d865ae11917a09dcbf85863e13eb0f5b7/esphome/components/adc/sensor.py#L21

so #ifndef USE_ADC_SENSOR_VCC should be similar

Reply:
@glmnet should I change it to:
`#ifdef USE_ADC_SENSOR_VCC`
?

Reply:
Yes that’s simpler and more consistent. 

Thanks